### PR TITLE
Admin: Organization summary export

### PR DESCRIPTION
**Pull request by sampoder**
## Summary of the problem

See #5066.

## Describe your changes

Added an additional export to `/admin/balances`

I would like feedback on addresses (we don't seem to be using this field anymore, do we even store addresses?) and the month by month revenue because I couldn't figure out a clean way to do that.

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -985,16 +986,34 @@ def balances
...
+               else
+                 [
+                   [:organization_id, ->(e) { e.id }],
+                   [:organization_name, ->(e) { e.name }],
+                   [:net_balance, ->(e) { render_balance.call(e, :settled_balance_cents) }],
+                   [:expenses, ->(e) { render_balance.call(e, :settled_outgoing_balance_cents) }],
+                   [:revenue, ->(e) { render_balance.call(e, :settled_incoming_balance_cents) }],
+                   [:category, ->(e) { e.category }],
+                   [:address, ->(e) { e.owner_address }], # are we using this? where are addresses stored?
+                   [:tags, ->(e) { e.event_tags.pluck(:name) }],
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]:
```suggestion
                   [:tags, ->(e) { e.event_tags.pluck(:name).join(",") }],
```

File: app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -975,6 +975,7 @@ def bookkeeping
   def balances
     @start_date = params[:start_date].present? ? Date.parse(params[:start_date]) : nil
     @end_date = params[:end_date].present? ? Date.parse(params[:end_date]) : nil
+    @partnered_events = params[:partnered_events] || false
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]: Remove scope. We want to export for all

**Comment by garyhtou** [comment]
We'll also need to remove that CSV file from github

**Comment by sampoder** [comment]
NEED TO ADD: by-month raised. use spread operator. use `settled_incoming_balance_cents`. use `min(event.created_at)`.

File: app/models/event.rb [hunk 3]
```diff
@@ -458,6 +458,15 @@ def settled_incoming_balance_cents(start_date: nil, end_date: nil)
         ct.sum(:amount_cents)
       end
   end
+  
+  def monthly_revenue(month)
```
Comment (garyhtou) [inline app/models/event.rb]: This seems fairly similar to

https://github.com/hackclub/bank/blob/48ebaaddfcf9b060a0a238bce5ac5a64276bd59b/app/models/event.rb#L451

Maybe we can re-use that

File: app/controllers/admin_controller.rb [hunk 4]
```diff
@@ -975,6 +975,7 @@ def bookkeeping
   def balances
     @start_date = params[:start_date].present? ? Date.parse(params[:start_date]) : nil
     @end_date = params[:end_date].present? ? Date.parse(params[:end_date]) : nil
+    @partnership_info = params[:partnership_info] || false
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]:
```suggestion
    @monthly_breakdown = params[:monthly_breakdown] || false
```

File: app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -985,16 +986,57 @@ def balances
...
       [:organization_name, ->(e) { e.name }],
       [:net_balance, ->(e) { render_balance.call(e, :settled_balance_cents) }],
       [:expenses, ->(e) { render_balance.call(e, :settled_outgoing_balance_cents) }],
       [:revenue, ->(e) { render_balance.call(e, :settled_incoming_balance_cents) }],
-      [:start_date, ->(_) { @start_date }],
-      [:end_date, ->(_) { @end_date }]
     ]
+
+
+    if @partnership_info
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]:
```suggestion
    if @monthly_breakdown
```

File: app/controllers/admin_controller.rb [hunk 6]
```diff
@@ -985,16 +986,57 @@ def balances
     render_balance = ->(event, type) {
       ApplicationController.helpers.render_money(event.send(type, start_date: @start_date, end_date: @end_date))
     }
+
+    render_monthly_revenue = ->(event, year, month) {
+      ApplicationController.helpers.render_money(
+        @monthly_revenue[event.id] ? @monthly_revenue[event.id][Date.new(year, month, 1).strftime("%Y-%m")] : 0
+      )
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]:
```suggestion
      key = Date.new(year, month, 1).strftime("%Y-%m")
      ApplicationController.helpers.render_money(
        @monthly_revenue.dig(event.id, key) || 0
      )
```

File: app/controllers/admin_controller.rb [hunk 7]
```diff
@@ -985,16 +986,57 @@ def balances
...
-      [:start_date, ->(_) { @start_date }],
-      [:end_date, ->(_) { @end_date }]
     ]
+
+
+    if @partnership_info
+      template.concat(
+        [
+          [:category, ->(e) { e.category }],
+          [:address, ->(e) { e.owner_address }], # are we using this? where are addresses stored?
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]:
```suggestion
```

File: app/controllers/admin_controller.rb [hunk 8]
```diff
@@ -985,16 +986,57 @@ def balances
...
+          [:tags, ->(e) { e.event_tags.pluck(:name).join(",") }],
+          [:joined, ->(e) { e.activated_at || e.created_at }],
+          [:start_date, ->(_) { @start_date }],
+          [:end_date, ->(_) { @end_date }]
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]: move this to all

File: app/controllers/admin_controller.rb [hunk 9]
```diff
@@ -985,16 +986,57 @@ def balances
...
+        ]
+      )
+      @monthly_revenue ||=
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]: replace `||=` with `=`

File: app/controllers/admin_controller.rb [hunk 10]
```diff
@@ -985,16 +986,57 @@ def balances
...
+                                          .where("canonical_transactions.amount_cents > ?", 0)
+                                          .group("canonical_event_mappings.event_id, date_trunc('month', canonical_transactions.created_at)")
+                                          .select("date_trunc('month', canonical_transactions.created_at) AS month,
+                               COALESCE(SUM(canonical_transactions.amount_cents), 0) AS revenue,
+                               canonical_event_mappings.event_id AS event_id")
+                                          .to_sql
+          ActiveRecord::Base.connection.exec_query(sql_query).each_with_object({}) do |item, result|
+            result[item["event_id"]] = {
+              item["month"].strftime("%Y-%m") => item["revenue"].to_i
+            }
```
Comment (garyhtou) [inline app/controllers/admin_controller.rb]:
```suggestion
            result[item["event_id"]] ||= {}
            result[item["event_id"]][item["month"]] = item["revenue"].to_i
```

**Review by garyhtou — COMMENTED** [review]
Preload event tag association

On line 984
```diff
-@events = filtered_events
+@events = filtered_events.includes(:event_tags)
```

File: app/views/admin/balances.html.erb [hunk 11]
```diff
@@ -25,6 +25,7 @@
         <span class='mr1'>EXPORT</span>
         <%= button_tag("CSV", value: "csv", name: "format", disable_with: false, formtarget: "_blank") %>
         <%= button_tag("JSON", value: "json", name: "format", disable_with: false, formtarget: "_blank") %>
+        <%= button_tag("CSV (partners)", value: "true", name: "partnership_info", type: "submit", formaction: "/admin/balances.csv", disable_with: false, formtarget: "_blank") %>
```
Comment (sampoder) [inline app/views/admin/balances.html.erb]:
```suggestion
        <%= button_tag("CSV (partners)", value: "true", name: "monthly_report", type: "submit", formaction: "/admin/balances.csv", disable_with: false, formtarget: "_blank") %>
```

**Review by garyhtou — APPROVED** [review]

**Pull request merged**